### PR TITLE
Add support for creating certmanager certificates

### PR DESCRIPTION
**Pull request by OlleLarsson**
Example values that i have used:

```
certmanager:
  enabled: true
  
  ca:
    commonName: example.com
    organization:
    - elastisys
  
  elasticsearch:
    transport:
      commonName: nodes.example.com
      organization:
      - elastisys
    admin: 
      commonName: nodes.example.com
      organization:
      - elastisys
```

### CODE REVIEW

File: helm/opendistro-es/values.yaml [hunk 1]
```diff
@@ -11,6 +11,59 @@
...
+      commonName: example.com
+      duration: 21600h
+      renewBefore: 360h
+      keySize: 2048
+      usages:
+      - server auth
+      - client auth
+    transport:
+      enabled: true
+      secretName: es-transport
```
Comment (ewnetu) [inline helm/opendistro-es/values.yaml]: I believe these secret names are going to be used for the different secrets under `elasticsearch:` and `kibana:`. Maybe also add them there.  something like
`elasticsearch:
   ssl:
      transport:
        existingCertSecret: **es-transport**
`?

Comment (OlleLarsson) [reply helm/opendistro-es/values.yaml]: Not really sure i see the problem here. Can't kibana just mount the transport certificate? 
```
kibana
  ssl:
    elasticsearch:
      enabled: true
      existingCertSecret: es-transport
      existingCertSecretCertSubPath: elk-rest-crt.pem
      existingCertSecretKeySubPath: elk-rest-key.pem
      existingCertSecretRootCASubPath: elk-rest-root-ca.pem
```

Comment (ewnetu) [reply helm/opendistro-es/values.yaml]: There is no issue at all. What I was asking is that the secret names are not added there and would be nice to have them. for example currently the secret name for `existingCertSecret` is not specified in values.yaml as you can see below. And I was suggesting to add it (also for the others)
```
kibana
  ssl:
    elasticsearch:
      enabled: false
      existingCertSecret: 
      existingCertSecretCertSubPath: elk-rest-crt.pem
      existingCertSecretKeySubPath: elk-rest-key.pem
      existingCertSecretRootCASubPath: elk-rest-root-ca.pem
```

**Review by lentzi90 — COMMENTED** [review]
I think this looks good, but I haven't tested it.
One question that popped up while reviewing:
Why do we need a certificate for kibana in this way? Can we not use the normal letsencrypt certificate with Ingress?

**Comment by OlleLarsson** [comment]
> I think this looks good, but I haven't tested it.
> One question that popped up while reviewing:
> Why do we need a certificate for kibana in this way? Can we not use the normal letsencrypt certificate with Ingress?

Well, i guess the traffic is not encrypted between nginx and kibana, which i am not sure is a problem or not.

**Comment by lentzi90** [comment]
> > I think this looks good, but I haven't tested it.
> > One question that popped up while reviewing:
> > Why do we need a certificate for kibana in this way? Can we not use the normal letsencrypt certificate with Ingress?
> 
> Well, i guess the traffic is not encrypted between nginx and kibana, which i am not sure is a problem or not.

For compliance, we probably want it at some point, but we are not doing it with any other services at the moment. I'd say skip it for now. When we get to the point when we want mTLS, let's install a service mesh and enable it for all services at once. :slightly_smiling_face:

File: helm/opendistro-es/values.yaml (hunk 1 above)
Comment (OlleLarsson) [reply helm/opendistro-es/values.yaml]: So, you mean that we should not set any default names for the certificates, whereby they must be specified through user passed values?

**Comment by OlleLarsson** [comment]
> For compliance, we probably want it at some point, but we are not doing it with any other services at the moment. I'd say skip it for now. When we get to the point when we want mTLS, let's install a service mesh and enable it for all services at once.

Sure, sound resonable. Let's say that i do a port-forward to kibana, then the traffic will not be encrypted if we remove tls from kibana, i guess?

**Comment by OlleLarsson** [comment]
Should we then also not use tls on the rest layer of elastisearch?

File: helm/opendistro-es/values.yaml (hunk 1 above)
Comment (ewnetu) [reply helm/opendistro-es/values.yaml]: that or put the secret names in their respective configurations. e.g.,

```
kibana
  ssl:
    elasticsearch:
      enabled: false
      existingCertSecret: es-rest
      existingCertSecretCertSubPath: elk-rest-crt.pem
      existingCertSecretKeySubPath: elk-rest-key.pem
      existingCertSecretRootCASubPath: elk-rest-root-ca.pem
```

**Comment by lentzi90** [comment]
> Sure, sound resonable. Let's say that i do a port-forward to kibana, then the traffic will not be encrypted if we remove tls from kibana, i guess?

Yes that is correct.

> Should we then also not use tls on the rest layer of elastisearch?

If we can (and do) expose it through an Ingress I would say we should do TLS in the ingress controller. I guess this is what we currently do but I'm not certain.

**Comment by OlleLarsson** [comment]
> > Sure, sound resonable. Let's say that i do a port-forward to kibana, then the traffic will not be encrypted if we remove tls from kibana, i guess?
> 
> Yes that is correct.
> 
> > Should we then also not use tls on the rest layer of elastisearch?
> 
> If we can (and do) expose it through an Ingress I would say we should do TLS in the ingress controller. I guess this is what we currently do but I'm not certain.

Ok. Should i disable and leave the functionality in or should i remove it completely?
Edit: i'll just remove it alltogether

**Comment by OlleLarsson** [comment]
I have updated the pr now. I have removed the rest and kibana certificates.
Now the secrets name will be taken from the `existingCertSecret` in the ssl configuration for elasticsearch. No default names are provided.
```
secretName: {{ .Values.elasticsearch.ssl.transport.existingCertSecret }}
secretName: {{ .Values.elasticsearch.ssl.admin.existingCertSecret }}
```

**Review by ewnetu — APPROVED** [review]
Looks good!

**Pull request merged**